test/cat.jl

```diff
@@ -94,4 +94,36 @@ for (f, d) in zip((vcat, hcat), (1, 2))
             @test dimnames(f(a, nda)) == dimnames(cat(a, nda; dims=d))
         end
     end
+    @testset "more than 2 dimensions" begin
+        ndt = NamedDimsArray(rand(Int8,2,2,2), (:x, :y, :z))
+
+        @test dimnames(hcat(ndt, ndt, parent(ndt))) == (:x, :y, :z)
+        @test dimnames(cat(ndt, ndt, parent(ndt), dims=:y)) == (:x, :y, :z)
+
+        @test_throws Exception hcat(ndt, parent(ndt), permutedims(ndt, (3,2,1)))
+        @test_throws Exception cat(ndt, parent(ndt), permutedims(ndt, (3,2,1)), dims=2)
+
+    end
+    @testset "tricky cases" begin
+        r23 = rand(Int8, 2,3)
+        nda = NamedDimsArray(rand(2,3), (:x, :y))
+        ndv = NamedDimsArray(rand(2), (:x,))
+
+        @test dimnames(hcat(nda, ndv)) == (:x, :y)
+        @test dimnames(cat(nda, ndv, dims=2)) == (:x, :y)
+
+        @test dimnames(hcat(r23', nda', nda')) == (:y, :x)
+        @test dimnames(cat(r23', nda', nda', dims=2)) == (:y, :x)
+
+        @test_throws Exception hcat(nda, r23')
+        @test_throws Exception cat(nda, r23', dims=2)
+
+        @test size(cat(nda', r23', nda', dims=(:x, :y))) == (9,6)
+        @test dimnames(cat(nda', r23', nda', dims=(1,2))) == (:y, :x)
+
+        @test dimnames(cat(1:2, ndv, [5 6]', dims=:z)) == (:x, :_, :z)
+
+        vcat(pi, ndv) # does not at present have names
```

Comment:
Nice one, I'll open an issue for this

Reply:
I messed around a bit... but not totally sure it has to be supported! 
```
function Base.vcat(xs::Union{Number, NamedDimsVector}...)
    nonames = map(xs) do x
        x isa AbstractVector ? unname(x) : x
    end
    data = vcat(nonames...)
    allL = map(xs) do x
        x isa AbstractVector ? dimnames(x) : (:_,)
    end
    newL = unify_names(allL...)
    NamedDimsArray{newL}(data)
end
```

Reply:
I've added https://github.com/invenia/NamedDims.jl/issues/135, might be a good in for a new contributor